src/useWindowScroll.ts

```diff
@@ -9,10 +9,10 @@ export interface State {
 }
 
 const useWindowScroll = (): State => {
-  const [state, setState] = useRafState<State>({
+  const [state, setState] = useRafState<State>(() => ({
     x: isClient ? window.pageXOffset : 0,
     y: isClient ? window.pageYOffset : 0,
-  });
+  }));
 
   useEffect(() => {
     const handler = () => {
```

Comment:
There should probably be a check inside the handler now, to make sure that the value has actually changed. Otherwise components that use this hook will end up rendering twice upon mount.

Reply:
fixed, see this commit: https://github.com/streamich/react-use/pull/1700/commits/55fbf2ffd10f5cf430f4b4df85f242917c46944f